dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/ValidationUtils.java

```diff
@@ -99,24 +100,31 @@ public static List<Note> validateNotes(
     return notes;
   }
 
-  public static List<MetadataIdentifier> validateMandatoryDataValue(
+  public static List<MetadataIdentifier> validateDeletionMandatoryDataValue(
       ProgramStage programStage, Event event, List<MetadataIdentifier> mandatoryDataElements) {
-    List<MetadataIdentifier> notPresentMandatoryDataElements = Lists.newArrayList();
-
     if (!needsToValidateDataValues(event, programStage)) {
-      return notPresentMandatoryDataElements;
+      return List.of();
```

Comment:
out of the scope of this PR, but why do we skip the validation if the event status is `SCHEDULE`?

Reply:
A `SCHEDULE` event is created at some point in time but it is meant to happen later at some point, so when it is created it is kind of `empty` and it doesn't make sense to require mandatory data values at that point.

Reply:
and in that case, are covered if I create and event this way, and then I provide the values later on?
what I mean by that is, at the moment I provide the values, do we treat it as an import strategy `CREATE` again to make sure all mandatory fields are set?

Reply:
Oh no, you are right.
At the moment if you create a schedule event without the mandatory datavalues and later you just change the status of the event then you can have an empty mandatory datavalues were they are meant to be present.

Will need to check how we manage the transition from a status to another